includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -692,7 +695,14 @@ private function process_payment_with_deferred_intent( int $order_id ) {
 
 			return [
 				'result'   => 'success',
-				'redirect' => $this->get_return_url( $order ),
+				'redirect' => sprintf(
+					'#wc-stripe-confirm-%s:%s:%s:%s',
+					$payment_needed ? 'pi' : 'si',
+					$order_id,
+					$payment_intent->client_secret,
+					wp_create_nonce( 'wc_stripe_update_order_status_nonce' )
```

Comment:
Just a heads up @a-danae that this changes the return of the `process_payment_with_deferred_intent()` function you introduced in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2754. 

This `redirect` ⇒ `#wc-stripe-confirm-%s:%s:%s:%s` string matches what is returned from `process_payment_with_saved_payment_method()` and was necessary to make sure the `confirmCardPayment()` JS function can verify 3DS if necessary. 

This doesn't appear to cause any issues with the classic checkout or break and redirect to the order received page after a successful checkout. 

This change does add support for 3DS confirmation on the classic checkout using your deferred intent changes. 

I'm curious if you have any thoughts on this. Does this seem correct?

Reply:
Thanks for the heads up James! 🙌 

Yupyup, looks good. I see payments with regular cards going through as expected in the classic checkout. Just a couple of things I have doubts about:

I get a request failing with 400 in the classic checkout after being redirected to the "Order received" page. I don't get it in the base branch, so it'd probably be worth checking? I used the regular `4242424242424242` card.

| Store | Stripe dashboard |
|--------|--------|
| <img width="300" alt="image" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/41606954/9c70fbda-c500-4fe6-ba2b-e80ba51a451f"> | <img width="300" alt="image" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/41606954/f2c7028d-e42e-4732-af53-176f9c7c81cc"> | 

I'm probably missing it, but where are we using the`payment_method` property we return? Just curious.
> 'payment_method' => $payment_intent->payment_method,

Reply:
> 'payment_method' => $payment_intent->payment_method,

Ah thanks. I think that make just be copy paste fail. WooPayments includes the payment method ID in the return, and I think I originally added it to check if it was necessary and didnt remove it. 

I've removed it in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2784/commits/aa85cb7fb7af59a117940e98881eb03f84d5db16

> I get a request failing with 400 in the classic checkout after being redirected to the "Order received" page. 

I'll have a look into that.

Reply:
> >  I get a request failing with 400 in the classic checkout after being redirected to the "Order received" page.
>
> I'll have a look into that.

I think I worked out what was causing that. The `process_payment_with_deferred_intent()` function creates and confirms the intent, responding with the `#wc-stripe-confirm-%s:%s:%s:%s` redirect would then cause it to confirm the intent again on the checkout. 

In https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2784/commits/65618e50b4a6e3445455cadab31ec0d73cb87318 I've fixed that by making sure we only respond with the `#wc-stripe-confirm-...` redirect which will subsequently attempt confirmation on the checkout (SCA) if the payment intent requires further action.